File: 0035-search-insert-position/0035-search-insert-position.cpp
class Solution {
public:
    int searchInsert(vector<int>& nums, int target) {
        int l = nums.size();
        for(int i = 0;i<l;i++){
            if(nums[i]==target){
                return i;
            }else if(nums[i]>target){
                return i;
            }
        }
        return l;
    }
};

File: 0035-search-insert-position/0035-search-insert-position.java
class Solution {
    public int searchInsert(int[] nums, int target) {
        for(int i =0;i<nums.length;i++){
            if(nums[i]==target){
                return i;
            }
        }
        if(target>nums[nums.length-1]){
            return nums.length;
        }else if(target<nums[0]){
            return 0;
        }else{
            for(int i = 0;i<nums.length-1;i++){
                if(nums[i]<=target && nums[i+1]>=target){
                    return i+1;
                }
            }
            return 0;
        }
    }
}